### DISCUSSION TITLE
Preferable island recognition methods

### DISCUSSION THREAD

**Discussion by reggi** [freshframework/fresh#877] [General]
I think that having the island folder is really limiting.

Instead someone else floated the idea of `file.island.tsx`.

OR

I'm assuming you could detect the node like this:

```tsx
const MyComponent = () => {
   return <div>hi</div>
}
MyComponent.island = true
```

If that's true you could Borrow from solid start, don't add the `island prop` and add it via special import:

```
const MyComponent = getIsland(() => import('../MyComponent'))

const UseIsland = () => {
   return <MyComponent/>
}
```

**Comment by marvinhagemeister** [comment]
The upcoming Fresh 1.4 version supports co-locating islands in the ´routes/` folder by putting islands in a folder named `(_islands)`. The PR that added this is here #1600 .
